examples/stdlib/stream/sum_of_squares.effekt

```diff
@@ -0,0 +1,12 @@
+import stream
+
+def squares(max: Int): Unit / emit[Int] = {
+  with val n = for[Int] { range(1, max) }
+  do emit(n * n)
+}
+
+def main() = {
+  val max = 10
+  println("The sum of squares from 1 to " ++ show(max) ++ " is:")
+  println(sum { squares(max + 1) })
+}
```

Comment:
This would be especially nice as an infinite stream, I think.
I tried `rangeFrom` and `take`, but it didn't seem very ergonomic to me :/

Reply:
Example is now
```
def squares(): Unit / emit[Int] =
  for[Int] { rangeFrom(1) } { n =>
    do emit(n * n)
  }


def main() = {
  val max = 10
  println("The sum of squares from 1 to " ++ show(max) ++ " is:")
  println(sum {
    with limit[Int](max + 1)
    squares()
  })
}
```
going through an infinite stream. I will rename `limit` back to `take`.